src/Latte/Engine.php

```diff
@@ -215,7 +215,7 @@ public function getCacheFile(string $name): string
 			? preg_replace('#[^\w@.-]+#', '-', substr($m[0], 1)) . '--'
 			: '';
 		$file = "{$this->tempDirectory}/{$base}{$hash}.php";
-		return file_exists($file) ? $file : false;
```

Comment:
IMHO,  `null` and `?string` return type is better.

Reply:
What are your thoughts on checking for the existence of the file before trying to read the file?

I'm surprised by the use of the silence @ operator throughout, and in my enviro (and likely some others), it still generates errors due to how the framework in use handles such errors, even when silenced.

Reply:
Sorry, I overlooked that you added `file_exists()` code. That's wrong. Method only builds the cache file name.